src/app-layout/drawer/index.tsx

```diff
@@ -131,47 +134,136 @@ export const Drawer = React.forwardRef(
   }
 );
 
-export function DrawerTriggersBar({
+export const DrawerTriggersBar = ({
   isMobile,
   topOffset,
   bottomOffset,
   drawers,
   contentClassName,
   toggleClassName,
-}: DrawerTriggersBarProps) {
+}: DrawerTriggersBarProps) => {
+  const [containerHeight, triggersContainerRef] = useContainerQuery(rect => rect.contentBoxHeight);
+  const isCompactMode = useCompactMode();
```

Comment:
There is already an existing hook: https://github.com/cloudscape-design/component-toolkit/blob/main/src/internal/visual-mode/index.ts#L31

It just lives in another package to allow reuse for border components:

```js
import { useDensityMode } from '@cloudscape-design/component-toolkit/internal';
```

Reply:
I also have an alternative approach, which does not need complex maths.

We can render a 2nd hidden copy of the drawers bar which does not have overflow checks. Then we take the height of the hidden bar and compare with the available height and make adjustments.

The alternative approach has an upside that this can be a single code for all modes and visual refresh

Reply:
This isn't complex - its one value for Comfortable mode, and one for Compact mode (In VR compact mode it doesn't actually change, so its 3 total values). I feel like rendering a hidden drawer is a bit overkill here